C/services/south-plugin-interfaces/python/python_plugin_interface.cpp

```diff
@@ -611,15 +611,15 @@ static void logErrorMessage()
 	//Get error message
 	PyObject *pType, *pValue, *pTraceback;
 	PyErr_Fetch(&pType, &pValue, &pTraceback);
+	PyErr_NormalizeException(&pType, &pValue, &pTraceback);
 
-	// NOTE from :
-	// https://docs.python.org/2/c-api/exceptions.html
-	//
-	// The value and traceback object may be NULL
-	// even when the type object is not.	
-	const char* pErrorMessage = pValue ?
-				    PyBytes_AsString(pValue) :
-				    "no error description.";
+	PyObject* str_exc_type = PyObject_Repr(pType);
+	PyObject* pyStr = PyUnicode_AsEncodedString(str_exc_type, "utf-8", "Error ~");
+	const char *strExcType =  PyBytes_AS_STRING(pyStr);
+
+	PyObject* str_exc_value = PyObject_Repr(pValue);
+	PyObject* pyExcValueStr = PyUnicode_AsEncodedString(str_exc_value, "utf-8", "Error ~");
+	const char *pErrorMessage =  PyBytes_AS_STRING(pyExcValueStr);
```

Comment:
Can you try replacing above 3 lines with just "char *pErrorMessage = PyString_AsString(pvalue);"

Reply:
Not working. Got below error:
```
Feb 22 13:08:46 asinha-ThinkPad-E450 FogLAMP[29381] INFO: modbustcp: foglamp.plugins.south.modbustcp.modbustcp: Modbus TCP Client is connected: False, 10.2.5.17:502
Feb 22 13:08:46 asinha-ThinkPad-E450 FogLAMP[29381] ERROR: modbustcp: foglamp.plugins.south.modbustcp.modbustcp: Failed to read data from modbus device. Got error the JSON object must be str, not 'dict'
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: ERROR: Called python script method plugin_poll : error while getting result object
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: logErrorMessage: Error 'TypeError("the JSON object must be str, not 'dict'",)'
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: Signal 11 (Segmentation fault) trapped:
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (0) 0             0x413a2d handler(int) + 61---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (1) 1       0x7f21039b74b0 /lib/x86_64-linux-gnu/libc.so.6(+0x354b0) [0x7f21039b74b0]---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (2) 2       0x7f2101cae8c0 PyFrame_GetLineNumber + 0---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (3) 3       0x7f2102268cf5 /home/asinha/dev/FogLAMP/cmake_build/C/services/south/../../lib/libsouth-plugin-python-interface.so(+0x2cf5) [0x7f2102268cf5]---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (4) 4       0x7f210226a0cc plugin_poll_fn + 1052---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (5) 5             0x4206a0 SouthPlugin::pollV2() + 32---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (6) 6             0x416bd9 SouthService::start(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, unsigned short) + 3897---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (7) 7             0x412f40 main + 336---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (8) 8       0x7f21039a2830 __libc_start_main + 240---------
Feb 22 13:08:46 asinha-ThinkPad-E450 Modbus TCP[29381]: FATAL: (9) 9             0x413559 _start + 41---------
```

So retained the earlier code.

Reply:
ok